Formula/colfer.rb

```diff
@@ -0,0 +1,17 @@
+class Colfer < Formula
+  desc "Schema compiler for binary data exchange"
+  homepage "https://github.com/pascaldekloe/colfer"
+  url "https://github.com/pascaldekloe/colfer/archive/v1.8.1.tar.gz"
+  sha256 "5d184c8a311543f26c957fff6cad3908b9b0a4d31e454bb7f254b300d004dca7"
+  license "CC0-1.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", bin/"colf", "./cmd/colf"
+  end
+
+  test do
+    shell_output("#{bin}/colf", 2)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.